airflow/providers/weaviate/hooks/weaviate.py

```diff
@@ -606,3 +636,303 @@ def object_exists(self, uuid: str | UUID, **kwargs) -> bool:
+
+        if isinstance(data, list):
+            data = pd.json_normalize(data)
+
+        unique_columns = sorted(data.columns.to_list())
+
+        self.log.info("Inserting %s objects.", data.shape[0])
+
+        if uuid_column is None or uuid_column not in data.columns:
+            (
+                data,
+                uuid_column,
+            ) = self._generate_uuids(
+                df=data,
+                class_name=class_name,
+                unique_columns=unique_columns,
+                vector_column=vector_column,
+                uuid_column=uuid_column,
+            )
+
+        # drop duplicate rows, using uuid_column and unique_columns. Removed  `None` as it can be added to
+        # set when `uuid_column` is None.
+        data = data.drop_duplicates(subset=[document_column, uuid_column], keep="first")
+        batch_delete_error: list = []
+        existing_documents, non_existing_documents = self._check_existing_documents(
+            data=data,
+            document_column=document_column,
+            uuid_column=uuid_column,
+            class_name=class_name,
+        )
+        if existing == "error" and len(existing_documents):
+            raise ValueError(
+                f"Documents {', '.join(existing_documents)} already exists. You can either skip or replace"
+                f" them by passing 'existing=skip' or 'existing=replace' respectively."
+            )
+        elif existing == "skip":
+            data = data[data[document_column].isin(non_existing_documents)]
+        elif existing == "replace":
+            batch_delete_error = self._delete_all_documents_objects(
+                document_keys=list(existing_documents),
+                document_column=document_column,
+                class_name=class_name,
+                batch_delete_error=batch_delete_error,
+                tenant=tenant,
+                batch_config_params=batch_config_params,
+            )
+            data = data[data[document_column].isin(non_existing_documents.union(existing_documents))]
+
+        insertion_errors: list = []
+        if data.shape[0]:
+            self.log.info("Batch inserting %s objects.", data.shape[0])
+            insertion_errors = self.batch_data(
+                class_name=class_name,
+                data=data,
+                insertion_errors=insertion_errors,
+                batch_config_params=batch_config_params,
+                vector_col=vector_column,
+                uuid_col=uuid_column,
+                tenant=tenant,
+            )
+            if insertion_errors or batch_delete_error:
+                if insertion_errors:
+                    self.log.info("Failed to insert %s objects.", len(insertion_errors))
+                if batch_delete_error:
+                    self.log.info("Failed to delete %s objects.", len(insertion_errors))
+                # Rollback object that were not created properly
+                self._delete_objects(
```

Comment:
shouldn't this be called rollback then?

Reply:
I think it should be `_delete_objects` only, since it is more specific and `rollback` is a little vague. IMO. But If you strongly feel about it, I can change it as well.